src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -240,6 +245,42 @@
 
     // @formatter:on
 
+    /**
+     * Overwritten due to pass-by-value nature of Java
+     * @author Maxqia
+     */
+    @Overwrite
+    public void setFire(int seconds) {
+        int ticks = seconds * 20;
+
+        if (((Object)this) instanceof EntityLivingBase) {
+            ticks = EnchantmentProtection.getFireTimeForEntity((EntityLivingBase) (Object) this, ticks);
+        }
+
+        if (!world.isRemote) {
+            PhaseContext context = ((IMixinWorldServer) world).getCauseTracker().getCurrentContext();
```

Comment:
What are the comments for saying "Here is where sponge is making a change.... This other part is vanilla code."

Look at other examples of overwrites (MixinWorld or MixinChunk even).